Get rid of "unused variable" warnings

cbuescher opened the pull request:
Another clean up trying to get rid of some annoying warnings. One large group that is
reasonably easy to tackle is the "unused variable" warnings. There are several cases
were most likely want to suppress the warnings (especially in the client documentation test
where the snippets should contain unused variable names). Most of the things can just be deleted though.

Commit:
WIP remove unused variable warnings

cbuescher commented on modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SSource.java:
```diff
@@ -130,7 +130,8 @@ public int getMaxLoopCounter() {
     private final String name;
-    private final String source;
+    // TODO can this be removed all up the call chain?
+    // private final String source;
```
@jdconrad I couldn't find any reference to this field, but maybe its here for debugging purposes or anything the like? In that case I would add a comment instead of removing. If this is really unused, maybe it can also be removed from the method signatures passing it down here. Could you take a look and comment?

jdconrad replied:
Will take a look!  Thanks for pointing this out.

jdconrad replied:
@cbuescher Looks like this is now extraneous.  It never got removed once source was set in the class above.  Please feel free to remove this as part of this PR or if you'd prefer I can remove it in a different PR.

cbuescher replied:
will do, thanks

Commit:
Remove TODO

This is most likey a leftover from 24d10ad and can go.

elasticmachine commented:
Pinging @elastic/es-core-infra

cbuescher commented on server/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java:
```diff
@@ -597,8 +600,9 @@ public void testPrimaryReference() throws Exception {
-        Request replicaRequest = (Request) primary.perform(request).replicaRequest;
+        // TODO check if the following two lines are really unused or might have side effects
+        // final Request request = new Request();
+        // Request replicaRequest = (Request) primary.perform(request).replicaRequest;
```
@ywelsch replicaRequest was used to test the primaryTerm() with "assertThat(replicaRequest.primaryTerm(), equalTo(primaryTerm));". Since that got removed, this part of the test can also go I guess. Does the test still perform its function after this? After all the primaryTerm gets mocked out somewhere above, but never asserted anywhere later? Could you take a look and comment?

ywelsch replied:
you can delete this test. It doesn't do anything meaningful.

Commit:
iter

rjernst commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java:
```diff
@@ -100,6 +100,7 @@
  */
 public class CRUDDocumentationIT extends ESRestHighLevelClientTestCase {
 
+    @SuppressWarnings("unused")
```
Where is the actual unused variable here?

cbuescher replied:
L164-L167 in the "tag::index-response" block. All the variables mentioned there should be in the documentation snippet but are unused. Same in most other documentation tests.

rjernst replied:
Could a dummy line be added to the snippet that "uses" the variables?

cbuescher replied:
That would be a lot of dummy lines, already five in this method, more in other test. I'm very much in favour of supressing them on the method in these particular cases.

Commit:
Remove version variables, leftovers from 6861d357

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
ScoreMode isn't used anymore, its CombineFunctions now

Commit:
iter

Commit:
iter

Commit:
Remove loop that is never going to be executed

Commit:
Remove unused params from SSource and Walker

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

cbuescher commented:
All the open questions are adressed now, I think this is ready for a full review now. Spin off in #31935 can be merged separately but is also included in this PR.

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

cbuescher commented:
I think this is ready for a final review now, all the spin-offs are taken care of.

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Re-add accidentally removed import

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

 Conflicts:
	client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SnapshotClientDocumentationIT.java
	server/src/test/java/org/elasticsearch/index/query/GeoDistanceQueryBuilderTests.java
	x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/RollupRequestTranslationTests.java

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

Commit:
Merge branch 'master' into remove-unused-variables

danielmitterdorfer reviewed (commented):
Thanks for updating the PR. I left a couple of comments.

cbuescher commented on libs/grok/src/main/java/org/elasticsearch/grok/Grok.java:
```diff
@@ -184,6 +184,7 @@ public String toRegex(String grokPattern) {
             String namedPatternRef = groupMatch(NAME_GROUP, region, grokPattern);
             String subName = groupMatch(SUBNAME_GROUP, region, grokPattern);
             // TODO(tal): Support definitions
```
Agreed, I just would like to keep the changes focused on the warnings with this PR.

danielmitterdorfer replied:
Yeah agreed, I also think we don't need to solve every problem here in this PR

cbuescher commented on libs/x-content/src/test/java/org/elasticsearch/common/xcontent/ObjectParserTests.java:
```diff
@@ -222,6 +222,7 @@ public void testDeprecationWarnings() throws IOException {
     public void testFailOnValueType() throws IOException {
         XContentParser parser = createParser(JsonXContent.jsonXContent, "{\"numeric_value\" : false}");
         class TestStruct {
+            @SuppressWarnings("unused")
```
Apparently the assignment in the lambda doesn't count as "use". If I remove the supression here I get a warning in Eclipse at least.

danielmitterdorfer replied:
Interesting, it's fine in IntelliJ. Then please disregard.

cbuescher commented on modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/KeepWordFilterFactory.java:
```diff
@@ -54,6 +54,7 @@
     private static final String KEEP_WORDS_KEY = "keep_words";
     private static final String KEEP_WORDS_PATH_KEY = KEEP_WORDS_KEY + "_path";
+    @SuppressWarnings("unused")
     private static final String KEEP_WORDS_CASE_KEY = KEEP_WORDS_KEY + "_case"; // for javadoc
```
The "_case" Setting is used in Analysis#getWordSet, but since that method also serves e.g. WordDelimiterTokenFilterFactory, it cannot use this concatenated constant. I think it still makes sense to keep it here for the docs though.

cbuescher commented on modules/parent-join/src/test/java/org/elasticsearch/join/aggregations/ChildrenIT.java:
```diff
@@ -458,13 +458,12 @@ public void testPostCollectAllLeafReaders() throws Exception {
 
         private Control(String category) {
-            this.category = category;
+            // category not used any further
```
Yes, makes sense to remove the whole constructor actually since the class is private already and only used inside this test class.

cbuescher commented on server/src/main/java/org/elasticsearch/common/lucene/search/XMoreLikeThis.java:
```diff
@@ -1026,30 +1026,20 @@ protected boolean lessThan(ScoreTerm a, ScoreTerm b) {
         }
 
-        void update(String word, String topField, float score, float idf, int docFreq, int tf) {
+        public void update(String word, String topField, float score) {
```
No, not sure how this happend. Removing.

cbuescher commented on server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java:
```diff
@@ -309,6 +309,7 @@ public void parse(ParseContext context) throws IOException {
                         double lat = context.parser().doubleValue();
                         token = context.parser().nextToken();
+                        @SuppressWarnings("unused")
                         Double alt = Double.NaN;
```
The later makes sense to me.

cbuescher commented on server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java:
```diff
@@ -309,6 +309,7 @@ public void parse(ParseContext context) throws IOException {
                         token = context.parser().nextToken();
```
Thats weird, doesn't show up in my IDE and token is definitly used here. Can you check again?

cbuescher commented on server/src/test/java/org/elasticsearch/cluster/metadata/TemplateUpgradeServiceTests.java:
```diff
@@ -298,7 +297,7 @@ public void testClusterStateUpdate() throws InterruptedException {
         }).when(mockIndicesAdminClient).deleteTemplate(any(DeleteIndexTemplateRequest.class), any(ActionListener.class));
 
-        final TemplateUpgradeService service = new TemplateUpgradeService(Settings.EMPTY, mockClient, clusterService, threadPool,
+        new TemplateUpgradeService(Settings.EMPTY, mockClient, clusterService, threadPool,
```
You are most likely right, I removed it and nothing bad happened (yet ;-)) Lets see what the tests bring up but I see this as a left-over as well.

cbuescher replied:
This brought the test to a griding halt, probably some side effects that make it necessary this is instantiated.

danielmitterdorfer replied:
Seems the reason is that `TemplateUpgradeService` has a side-effect: it registers itself as a cluster service listener. IMHO constructors should be free of such side effects and this belongs into some kind of start / post-construct method - but again: out of scope for this PR.

cbuescher commented on server/src/test/java/org/elasticsearch/common/util/concurrent/AsyncIOProcessorTests.java:
```diff
@@ -39,7 +39,7 @@ public void testPut() throws InterruptedException {
                 if (blockInternal) {
                     synchronized (this) {
-                        for (Tuple<Object, Consumer<Exception>> c :candidates) {
+                        for (int i = 0; i < candidates.size(); i++) {
```
Not sure, addAndGet is used in the "else" branch of the statement in L41, maybe its there for a purpose. I didn't go into the logic of the test too much to be honest, but I thought it better to keep it as is.

danielmitterdorfer replied:
Well, it adds 1 for `candidates.size()` times which is functionally identical to add it directly without the loop.

cbuescher replied:
I agree and I added a TODO to re-check this, but I would prefer not to change it with this PR since I don't fully understand the test. Might be we can remove it though.

danielmitterdorfer replied:
Sounds good to me. Thanks for adding this.

cbuescher commented on server/src/test/java/org/elasticsearch/search/aggregations/pipeline/moving/avg/MovAvgIT.java:
```diff
@@ -984,11 +980,12 @@ public void testTwoMovAvgsWithPredictions() {
 
     public void testBadModelParams() {
         try {
+            @SuppressWarnings("unused")
```
Actually it is not completely clear to me what the expectations are here. Currently it seems the test doesn't throw an Exception. Will check what it is supposed to do.

cbuescher replied:
Talked to the team and opened https://github.com/elastic/elasticsearch/issues/34046 to clarify this test, adding @AwaitsFix in the meantime

cbuescher commented on server/src/test/java/org/elasticsearch/search/suggest/SuggestSearchIT.java:
```diff
@@ -972,9 +972,7 @@ public void testSuggestWithManyCandidates() throws InterruptedException, Executi
         searchSuggest = searchSuggest("united states house of representatives elections in washington 2006", "title", suggest);
-        long total = System.currentTimeMillis() - start;
         assertSuggestion(searchSuggest, 0, 0, "title", "united states house of representatives elections in washington 2006");
         // assertThat(total, lessThan(1000L)); // Takes many seconds without fix - just for debugging
```
+1

cbuescher commented on server/src/test/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java:
```diff
@@ -55,7 +55,6 @@ public void testThreadNames() throws Exception {
             }
         }
         logger.info("pre node threads are {}", preNodeStartThreadNames);
-        String node = internalCluster().startNode();
```
Yes, good point, will add that back.

cbuescher commented on x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java:
```diff
@@ -177,6 +177,7 @@ public LogicalPlan visitRelation(RelationContext ctx) {
     private Join doJoin(LogicalPlan left, JoinRelationContext ctx) {
```
I agree but would prefer to do this in a separate PR to get a review from the SQL guys, even if its most likely dead code.

cbuescher commented on x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/SourceGeneratorTests.java:
```diff
@@ -64,6 +64,7 @@ public void testLimit() {
         SearchSourceBuilder sourceBuilder = SourceGenerator.sourceBuilder(container, null, randomIntBetween(1, 10));
         Builder aggBuilder = sourceBuilder.aggregations();
         assertEquals(1, aggBuilder.count());
+        @SuppressWarnings("unused")
```
Good point.

cbuescher commented on x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/pagerduty/PagerDutyAccount.java:
```diff
@@ -29,15 +29,13 @@
     private final String serviceKey;
     private final HttpClient httpClient;
     private final IncidentEventDefaults eventDefaults;
-    private final Logger logger;
```
done

cbuescher commented on x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/test/AbstractWatcherIntegrationTestCase.java:
```diff
@@ -541,12 +541,10 @@ public EmailSent send(Email email, Authentication auth, Profile profile, String
 
         private final List<ScheduleTriggerEngineMock> schedulers;
         private final ClockMock clock;
-        private final Logger logger;
```
done

Commit:
iteration: adressing comments

cbuescher commented on test/framework/src/main/java/org/elasticsearch/test/store/MockFSDirectoryService.java:
```diff
@@ -87,9 +86,6 @@ public MockFSDirectoryService(IndexSettings idxSettings, IndexStore indexStore,
 
         randomIOExceptionRate = RANDOM_IO_EXCEPTION_RATE_SETTING.get(indexSettings);
         randomIOExceptionRateOnOpen = RANDOM_IO_EXCEPTION_RATE_ON_OPEN_SETTING.get(indexSettings);
-        preventDoubleWrite = RANDOM_PREVENT_DOUBLE_WRITE_SETTING.get(indexSettings);
```
did that.

cbuescher commented:
@danielmitterdorfer thanks a lot for getting back to this. I addressed most of your comments with the latest commit, checking back on how MovAvgIT is supposed to work. Could you go through the other comments I replied to and see if you agree?

Commit:
Muting MovAvgIT#testBadModelParams

danielmitterdorfer commented on server/src/test/java/org/elasticsearch/index/query/GeoDistanceQueryBuilderTests.java:
```diff
@@ -298,10 +298,10 @@ public void testParsingAndToQuery12() throws IOException {
+            throws IOException {
         parseQuery(query).toQuery(createShardContext());
-     // TODO: what can we check?
+        // TODO: what can we check? See https://github.com/elastic/elasticsearch/issues/34043
```
Thanks for creating the issue!

Commit:
iter

danielmitterdorfer reviewed (approved):
Thanks @cbuescher! LGTM

Commit:
Revert removing code in TemplateUpdateServiceTests

Commit:
Merge branch 'master' into remove-unused-variables

s1monw commented:
@cbuescher I think this change caused some failures I am hitting locally. It reproduces on master:

```
./gradlew :server:integTest -Dtests.seed=CE15410BFF4D8B8A -Dtests.class=org.elasticsearch.search.aggregations.pipeline.moving.avg.MovAvgIT -Dtests.method="testHoltWintersNotEnoughData" -Dtests.security.manager=true -Dtests.locale=fi -Dtests.timezone=JST -Dcompiler.java=10 -Druntime.java=8
```

```
2> REPRODUCE WITH: ./gradlew :server:integTest -Dtests.seed=CE15410BFF4D8B8A -Dtests.class=org.elasticsearch.search.aggregations.pipeline.moving.avg.MovAvgIT -Dtests.method="testHoltWintersNotEnoughData" -Dtests.security.manager=true -Dtests.locale=fi -Dtests.timezone=JST -Dcompiler.java=10 -Druntime.java=8
FAILURE 0.39s | MovAvgIT.testHoltWintersNotEnoughData <<< FAILURES!
   > Throwable #1: junit.framework.AssertionFailedError: Expected exception SearchPhaseExecutionException but no exception was thrown
   >    at __randomizedtesting.SeedInfo.seed([CE15410BFF4D8B8A:A5C013FEF4A98B75]:0)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2687)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2672)
   >    at org.elasticsearch.search.aggregations.pipeline.moving.avg.MovAvgIT.testHoltWintersNotEnoughData(MovAvgIT.java:856)
   >    at java.lang.Thread.run(Thread.java:748)
```

s1monw commented:
I opened #34098
